docs/anomaly-detection/components/models.md

```diff
@@ -644,38 +646,46 @@ logger = logging.getLogger(__name__)
 
 
 class CustomModel(Model):
-    """
-    Custom model implementation.
-    """
-
-    # by default, each `Model` will be created as a univariate one
-    # uncomment line below for it to be of multivariate type
-    # is_multivariate = True
-
-    def __init__(self, percentage: float = 0.95, **kwargs):
-        super().__init__(**kwargs)
-        self.percentage = percentage
-        self._mean = np.nan
-        self._std = np.nan
-
-    def fit(self, df: pd.DataFrame):
-        # Model fit process: 
-        y = df['y']
-        self._mean = np.mean(y)
-        self._std = np.std(y)
-        if self._std == 0.0:
-            self._std = 1 / 65536
-
-    def infer(self, df: pd.DataFrame) -> np.array:
-        # Inference process:
-        y = df['y']
-        zscores = (y - self._mean) / self._std
-        anomaly_score_cdf = st.norm.cdf(np.abs(zscores))
-        df_pred = df[['timestamp', 'y']].copy()
-        df_pred['anomaly_score'] = anomaly_score_cdf > self.percentage
-        df_pred['anomaly_score'] = df_pred['anomaly_score'].astype('int32', errors='ignore')
-
-        return df_pred
+  """
+  Custom model implementation.
+  
+  by default, each `Model` will be created as a univariate one
+  uncomment line below for it to be of multivariate type
+  `is_multivariate = True`
+  
+  """
+  
+  def __init__(self, percentage: float = 0.95, **kwargs):
+    super().__init__(**kwargs)
+    self.percentage = percentage
+    self._mean = np.nan
+    self._std = np.nan
+
+  def fit(self, df: pd.DataFrame):
+    # Model fit process:
+    y = df['y']
+    self._mean = np.mean(y)
+    self._std = np.std(y)
+    if self._std == 0.0:
+      self._std = 1 / 65536
+
+  def infer(self, df: pd.DataFrame) -> np.array:
+    # Inference process:
+    y = df['y']
+    zscores = (y - self._mean) / self._std
+    anomaly_score_cdf = st.norm.cdf(np.abs(zscores))
+    df_pred = df[['timestamp', 'y']].copy()
+    df_pred['anomaly_score'] = anomaly_score_cdf > self.percentage
+    df_pred['anomaly_score'] = df_pred['anomaly_score'].astype('int32', errors='ignore')
+
+    return df_pred
+
+  def serialize(self) -> None:
+    pass
```

Comment:
It's better to use pickle.dumps() here in example so it works for both in-memory and on-disk model storage

The same with pickle.loads() here in `deserialize`. See internal models example if needed